### ISSUE TITLE
Capture dialog hangs at "Saving data" for some invalid filenames

### ISSUE THREAD

**Issue by Golddouble** [danny0838/webscrapbook#308]
Hello,

I was able to capture a lot of websites with this amazing addon. But here is a site that is stuck.

I was not able to capture the following webpage, even if I have tried while I was logged out:
https://forum.mxlinux.org/viewtopic.php?t=70730

It just hangs:
![k20221210-222830](https://user-images.githubusercontent.com/13977359/206875914-bcc61ef0-d613-4dfc-a42c-cf9c1ec8fdb5.png)

My settings:
![k20221210-222457](https://user-images.githubusercontent.com/13977359/206875953-deb78f14-b817-458c-83fc-fdbff789ebac.png)
![k20221210-222514](https://user-images.githubusercontent.com/13977359/206875952-9d8dcab1-5fd8-4041-80c6-27766717ada7.png)
![k20221210-222341](https://user-images.githubusercontent.com/13977359/206875954-e029f2ca-ea9f-4d0c-b4aa-41834d54fe54.png)

Can you reproduce this also by being logged out? Thank you.

**Comment by danny0838** [comment]
Please provide full options. You can usually copy it from `Capture as` => `Advanced`

**Comment by Golddouble** [comment]
```
{
 "tasks": [
  {
   "tabId": 2,
   "url": "https://forum.mxlinux.org/viewtopic.php?t=70730",
   "title": "Since yesterday: When I create a print job, the printer does not do anything, however: scanning works - MX Linux Forum",
   "comment": ""
  }
 ],
 "bookId": null,
 "parentId": "root",
 "index": null,
 "delay": null,
 "options": {
  "capture.serverUploadWorkers": 4,
  "capture.serverUploadRetryCount": 2,
  "capture.serverUploadRetryDelay": 2000,
  "capture.saveTo": "folder",
  "capture.saveFolder": "MX-Forum-Archiv (symlink)",
  "capture.saveAs": "folder",
  "capture.saveFilename": "%title%; archiviert am %create-m%.%create-Y%",
  "capture.saveFilenameMaxLenUtf16": 120,
  "capture.saveFilenameMaxLenUtf8": 240,
  "capture.saveAsciiFilename": false,
  "capture.saveOverwrite": false,
  "capture.saveFileAsHtml": false,
  "capture.saveDataUriAsFile": true,
  "capture.saveDataUriAsSrcdoc": true,
  "capture.saveResourcesSequentially": false,
  "capture.resourceSizeLimit": null,
  "capture.image": "save",
  "capture.imageBackground": "save-used",
  "capture.favicon": "save",
  "capture.faviconAttrs": "",
  "capture.canvas": "remove",
  "capture.audio": "blank",
  "capture.video": "blank",
  "capture.embed": "save",
  "capture.object": "save",
  "capture.applet": "blank",
  "capture.frame": "save",
  "capture.frameRename": true,
  "capture.font": "save-used",
  "capture.style": "save",
  "capture.styleInline": "save",
  "capture.rewriteCss": "url",
  "capture.mergeCssResources": true,
  "capture.script": "remove",
  "capture.noscript": "blank",
  "capture.contentSecurityPolicy": "remove",
  "capture.ping": "blank",
  "capture.preload": "remove",
  "capture.prefetch": "remove",
  "capture.base": "blank",
  "capture.formStatus": "html-all",
  "capture.shadowDom": "save",
  "capture.removeHidden": "none",
  "capture.linkUnsavedUri": true,
  "capture.downLink.file.mode": "header",
  "capture.downLink.file.extFilter": "###image\nbmp, gif, ico, jpg, jpeg, jpe, jp2, png, tif, tiff, svg\n###audio\n#aac, ape, flac, mid, midi, mp3, ogg, oga, ra, ram, rm, rmx, wav, wma\n###video\n#avc, avi, flv, mkv, mov, mpg, mpeg, mp4, wmv\n###archive\nzip, rar, jar, bz2, gz, tar, rpm, 7z, 7zip, xz, jar, xpi, lzh, lha, lzma\n#/z[0-9]{2}|r[0-9]{2}/\n###document\npdf, doc, docx, xls, xlsx, ppt, pptx, odt, ods, odp, odg, odf, rtf, txt, csv\n###executable\n#exe, msi, dmg, bin, xpi, iso\n###any non-web-page\n#/(?!$|html?|xht(ml)?|php|py|pl|aspx?|cgi|jsp)(.*)/i",
  "capture.downLink.doc.depth": 1,
  "capture.downLink.doc.delay": null,
  "capture.downLink.doc.mode": "source",
  "capture.downLink.doc.urlFilter": "# /^https://forum\\.mxlinux\\.org/viewtopic\\.php\\?t\\=.*\\&start/",
  "capture.downLink.urlFilter": "###skip common logout URL\n/[/=]logout\\b/i\n# /[/=]viewtopic\\b/i\n/[/=]search\\b/i\n/[/=]memberlist\\b/i\n/[/=]index\\b/i\n/[/=]posting\\b/i\n/[/=]viewforum\\b/i\n\n/[/=]forum\\.mxlinux\\.org/app\\b/i\n/[/=]forum\\.mxlinux\\.org/ucp\\b/i\n/[/=]viewonline\\.php\\b/i\n\n/[/=]forum\\.mxlinux\\.org/index\\.php\\b/i\n/[/=]forum\\.mxlinux\\.org/viewtopic\\.php\\?uid\\b/i\n/[/=]forum\\.mxlinux\\.org/viewtopic\\.php\\?p\\=\\b/i\n/[/=]forum\\.mxlinux\\.org/viewtopic\\.php.*bookmark.*\\b/i\n/[/=]forum\\.mxlinux\\.org/viewtopic\\.php.*view\\=print.*\\b/i\n/[/=]forum\\.mxlinux\\.org/viewtopic\\.php\\?f\\=\\b/i\n\n/[/=]bugs\\.mxlinux\\.org\\b/i\n/[/=]github\\.com/MX-Linux\\b/i\n/[/=]zumaclub\\.ru\\b/i\n/[/=]www\\.phpbb\\.com\\b/i\n\n/[/=]localhost\\:631\\b/i\n",
  "capture.downLink.urlExtra": "",
  "capture.referrerPolicy": "strict-origin-when-cross-origin",
  "capture.referrerSpoofSource": false,
  "capture.recordDocumentMeta": true,
  "capture.recordRewrites": false,
  "capture.prettyPrint": false,
  "capture.insertInfoBar": false,
  "capture.helpersEnabled": false,
  "capture.helpers": "",
  "capture.remoteTabDelay": null,
  "capture.deleteErasedOnCapture": true,
  "capture.deleteErasedOnSave": false,
  "capture.backupForRecapture": true,
  "capture.zipCompressLevel": null
 }
}
```

**Comment by danny0838** [comment]
Seems caused by bad filename handling. Setting save filename back to `%id%` shoulf workaround the problem, before we come out a final fix.

**Comment by Golddouble** [comment]
Thank you for reproducing this. I was successful with your workaround.

**Comment by danny0838** [comment]
Fixed in 1.5.2
